qiskit/transpiler/passes/optimization/consolidate_blocks.py

```diff
@@ -78,9 +74,15 @@ def __init__(
         if kak_basis_gate is not None:
             self.decomposer = TwoQubitBasisDecomposer(kak_basis_gate)
         elif basis_gates is not None:
-            self.decomposer = unitary_synthesis._decomposer_2q_from_basis_gates(
-                basis_gates, approximation_degree=approximation_degree
-            )
+            kak_gates = set(basis_gates or []).intersection(KAK_GATE_NAMES.keys())
```

Comment:
You should be able to skip construction of the intermediate set by using the key view as the receiver. Not that it'd make much difference in this context 🙂.

```suggestion
            kak_gates = KAK_GATE_NAMES.keys() & (basis_gates or [])
```